llvm/test/Feature/Repo/repo_InsertOrExtraValueInst_hash.ll

```diff
@@ -22,6 +23,6 @@ define internal void @__cxx_global_var_init() section ".text.startup" personalit
 
 declare i32 @__gxx_personality_v0(...)
 
-;CHECK: !0 = !RepoDefinition(name: ".str",
-;CHECK-NEXT: !1 = !RepoDefinition(name: "__cxx_global_var_init",
+;CHECK: !0 = !RepoDefinition(name: {{.*}}, digest: [16 x i8] c"{{.*}}", linkage: external, visibility: default, pruned: true)
+;CHECK-NEXT: !1 = !RepoDefinition(name: {{.*}}, digest: [16 x i8] c"{{.*}}", linkage: external, visibility: default, pruned: true)
```

Comment:
Could you explain this change, please?

Reply:
This test is used to make sure that the 'insertvalue' instruction hash is calculated correctly and pruned in the second time compilation. However, the initial test didn't test `pruned` flag, therefore, I added the `prund` flag test.

Secondly, when fixing the issue of missing DCE pass , I found the order of the `repodefinition` metadata has changed from
```
!0 = !RepoDefinition(name: ".str",
!1 = !RepoDefinition(name: "__cxx_global_var_init",
```
to:
```
!0 = !RepoDefinition(name: "__cxx_global_var_init",
!1 = !RepoDefinition(name: ".str",
```

The test mainly focus on the number of the `repodefinition` metadata, therefore the `name` checks have been removed.